src/pages/tutor-profile/TutorProfile.jsx

```diff
@@ -41,20 +41,43 @@ const TutorProfile = () => {
     defaultResponse: defaultResponses.array
   })
 
+  useEffect(() => {
+    if (
+      !loading &&
+      response &&
+      response.videoLink &&
+      response.videoLink.tutor !== null &&
+      response.videoLink.tutor !== ''
+    ) {
+      setVideoPreview(false)
+    }
+  }, [loading, response])
+
   if (loading) {
     return <Loader pageLoad size={70} />
   }
 
   const isTutor = preferredRole === UserRoleEnum.Tutor
   const shouldShowPresentation =
     isTutor || (!isTutor && response.videoLink?.student)
-
+  const videoBlock =
+    userRole === tutor ? (
+      <VideoPresentation
+        video={response.videoLink.tutor}
+        videoPreview={videoPreview}
+      />
+    ) : (
+      <VideoPresentation
+        video={response.videoLink.student}
+        videoPreview={videoPreview}
+      />
+    )
```

Comment:
I think these components are pretty similar. I suggest refactoring it as following:

```suggestion   
    <VideoPresentation
        video={response.videoLink[condition here...}
        videoPreview={videoPreview}
      />
```

Also I noticed line of code `const isTutor = preferredRole === UserRoleEnum.Tutor`. Please consider reusing it or other code if possible

Reply:
Definitely!